vm/main.go

```diff
@@ -47,44 +49,107 @@ func listen(path string) (net.Listener, error) {
 
 // ready checks whether sqlcl is ready or not by querying localhost:9080.
 func ready(ctx echo.Context) error {
-	url := "http://localhost:7681/" // "sqlcl" is the name of the service defined in docker-compose.yml
-	resp, err := http.Get(url)
-	if err != nil {
-		log.Println(err)
-		return ctx.String(http.StatusOK, "false")
-
+	if ttyd.IsRunning() {
+		return ctx.String(http.StatusOK, "true")
 	}
-	defer resp.Body.Close()
-
-	return ctx.String(resp.StatusCode, "true")
-
-	// return ctx.JSON(http.StatusOK, HTTPMessageBody{Message: "hello from HTTP"})
 
+	return ctx.String(http.StatusServiceUnavailable, "false")
 }
 
 // dark starts ttyd in dark mode
 func dark(ctx echo.Context) error {
-	if !daemonStarted {
-		cmd := exec.Command("/usr/bin/ttyd", "-u", "1000", "-g", "1000", "-t", "titleFixed='sqlcl'", "/bin/bash", "/home/sql.sh")
-		cmd.Start()
-		daemonStarted = true
+	if err := ttyd.Start(DarkTheme); err != nil {
+		log.Printf("failed to start ttyd with dark mode: %s\n", err)
+
+		return echo.NewHTTPError(http.StatusInternalServerError)
 	}
-	return ctx.String(http.StatusOK, "true")
 
+	log.Println("started ttyd with dark mode")
+
+	return ctx.String(http.StatusOK, "true")
 }
 
 // light starts ttyd in dark mode
 func light(ctx echo.Context) error {
-	if !daemonStarted {
-		cmd := exec.Command("/usr/bin/ttyd", "-u", "1000", "-g", "1000", "-t", "titleFixed='sqlcl'", "-t", "theme={'background': '#ffffff', 'foreground': '#2b2b2b', 'cursor': '#adadad', 'selection': '#ddb6fc'}", "/bin/bash", "/home/sql.sh")
-		cmd.Start()
-		daemonStarted = true
+	if err := ttyd.Start(LightTheme); err != nil {
+		log.Printf("failed to start ttyd with light mode: %s\n", err)
+
+		return echo.NewHTTPError(http.StatusInternalServerError)
 	}
-	return ctx.String(http.StatusOK, "true")
 
+	log.Println("started ttyd with light mode")
+
+	return ctx.String(http.StatusOK, "true")
 }
 
 type HTTPMessageBody struct {
 	Message string `json:"message"`
 	Body    string `json:"body,omitempty"`
 }
+
+type Theme string
+
+const (
+	DarkTheme  Theme = "dark"
+	LightTheme Theme = "light"
+)
+
+type TTYD struct {
```

Comment:
I created a struct to wrap the command. So that the API is simple and it manages the TTYD instance (stop, start, check if running, etc.)